src/main/kotlin/uk/gov/justice/digital/hmpps/deliusapi/service/allocation/AllocationService.kt

```diff
@@ -0,0 +1,217 @@
+      startDate = newOffenderManager.allocationDate,
+      endDate = activeAtTheTimeResponsibleOfficer.endDate,
+      communityOffenderManager = newOffenderManager
+    )
+    responsibleOfficerRepository.save(newResponsibleOfficer)
+    activeAtTheTimeResponsibleOfficer.endDate = newOffenderManager.allocationDate
+    responsibleOfficerRepository.save(activeAtTheTimeResponsibleOfficer)
+    if (activeAtTheTimeResponsibleOfficer.communityOffenderManager != null &&
+      newResponsibleOfficer.communityOffenderManager!!.active
+    ) {
+      createResponsibleOfficerInternalTransferContact(offender, newResponsibleOfficer, activeAtTheTimeResponsibleOfficer)
+    }
+  }
+
+  private fun createInternalTransferContact(
+    offender: Offender,
+    newOffenderManager: OffenderManager,
+    oldManager: OffenderManager
+  ) {
+    systemContactService.createSystemContact(
+      NewSystemContact(
+        type = WellKnownContactType.INTERNAL_TRANSFER,
+        offenderId = offender.id,
+        date = newOffenderManager.allocationDate.toLocalDate(),
+        teamId = newOffenderManager.team.id,
+        staffId = newOffenderManager.staffAllocation.id,
+        providerId = newOffenderManager.provider.id,
+        notes =
+        """Transfer Reason: Internal Transfer
+          Transfer Date: ${newOffenderManager.allocationDate.format(DateTimeFormatter.ofPattern("dd/MM/yyyy"))}
+          From Trust: ${newOffenderManager.provider.description}
+          From Team: ${oldManager.team.description}
+          From Officer: ${oldManager.staffAllocation.getDisplayName()}
+          -------------------------------/${System.lineSeparator()}"""
+          .trimIndent()
+      )
+    )
+  }
+
+  private fun createResponsibleOfficerInternalTransferContact(
+    offender: Offender,
+    newResponsibleOfficer: ResponsibleOfficer,
+    oldResponsibleOfficer: ResponsibleOfficer
+  ) {
+    systemContactService.createSystemContact(
+      NewSystemContact(
+        type = WellKnownContactType.RESPONSIBLE_OFFICER_CHANGE,
+        offenderId = offender.id,
+        date = LocalDate.now(),
+        providerId = newResponsibleOfficer.communityOffenderManager!!.provider.id,
+        teamId = newResponsibleOfficer.communityOffenderManager!!.team.id,
+        staffId = newResponsibleOfficer.communityOffenderManager!!.staffAllocation.id,
+        notes =
+        """Transfer Reason: Internal Transfer
+          Transfer Date: ${newResponsibleOfficer.communityOffenderManager!!.allocationDate.format(DateTimeFormatter.ofPattern("dd/MM/yyyy"))}
+          From Trust: ${oldResponsibleOfficer.communityOffenderManager!!.provider.description}
+          From Team: ${oldResponsibleOfficer.communityOffenderManager!!.team.description}
+          From Officer: ${oldResponsibleOfficer.communityOffenderManager!!.staffAllocation.getDisplayName()}
+          -------------------------------/${System.lineSeparator()}"""
+          .trimIndent()
+      )
+    )
+  }
```

Comment:
Might be cleaner to do the null-checks at the start here?

```suggestion
  ) {
    val newCommunityOffenderManager = newResponsibleOfficer.communityOffenderManager!!
    val oldCommunityOffenderManager = oldResponsibleOfficer.communityOffenderManager!!
    systemContactService.createSystemContact(
      NewSystemContact(
        type = WellKnownContactType.RESPONSIBLE_OFFICER_CHANGE,
        offenderId = offender.id,
        date = LocalDate.now(),
        providerId = newCommunityOffenderManager.provider.id,
        teamId = newCommunityOffenderManager.team.id,
        staffId = newCommunityOffenderManager.staffAllocation.id,
        notes = """
          Transfer Reason: Internal Transfer
          Transfer Date: ${newCommunityOffenderManager.allocationDate.format(DateTimeFormatter.ofPattern("dd/MM/yyyy"))}
          From Trust: ${oldCommunityOffenderManager.provider.description}
          From Team: ${oldCommunityOffenderManager.team.description}
          From Officer: ${oldCommunityOffenderManager.staffAllocation.getDisplayName()}
          -------------------------------
          
          """.trimIndent()
      )
    )
  }
```